index.html

```diff
@@ -1,577 +1,194 @@
-    },
-    {
-      url: "https://exokit.org/",
-      title: "Home",
-      active: window.location.host.split(".")[0] === "exokit"
-    }
-  ]
 
-  document.getElementById('nav-popout').innerHTML = navLinks.map((link, index) => {
-    return `
-      <div>
-        <a href=${link.url} class=${link.active ? "activeNav" : ""}>${link.title}</a>
-      </div>
-    `
-  }).join(`<br/>`)
+          const ress = await Promise.all([
+            fetch('abi.json'),
+            fetch('address.json'),
+          ]);
+          const [jsonInterface, address] = await Promise.all(ress.map(res => res.json()));
+          const contract = web3.eth.contract(jsonInterface).at(address);
+          accept(contract);
+        });
+      });
+
+      const res = await fetch(`https://content.exokit.org/users/avaer`);
+      const root = await res.json();
+
+      const files = [];
+      const hashes = [];
+      let keypath = '';
+      const _recurse = node => {
+        keypath += (keypath ? '/' : '') + node.name;
+
+        if (node.hash) {
+          files.push(keypath);
+          hashes.push('0x' + node.hash);
+        }
+        if (node.children) {
+          for (let i = 0; i < node.children.length; i++) {
+            _recurse(node.children[i]);
+          }
+        }
+      };
+      _recurse(root);
+      const ids = await Promise.all(hashes.map(hash => new Promise((accept, reject) => {
+        // const hash = '0x' + filename.match(/([^\/]*)$/)[1];
+        contract.getId(hash, (err, id) => {
+          if (!err) {
+            accept(id.toNumber());
+          } else {
+            reject(err);
+          }
+        });
+      })));
+
+      // Define links for popout-nav
+      const navLinks = [
+        {
+          url: "https://store.exokit.org/",
+          title: "Store",
+          active: window.location.host.split(".")[0] === "store"
+        },
+        {
+          url: "https://avatars.exokit.org/",
+          title: "Avatars",
+          active: window.location.host.split(".")[0] === "avatars"
+        },
+        {
+          url: "https://multiplayer.exokit.org/",
+          title: "Multiplayer",
+          active: window.location.host.split(".")[0] === "multiplayer"
+        },
+        {
+          url: "https://editor.exokit.org/",
+          title: "Editor",
+          active: window.location.host.split(".")[0] === "editor"
+        },
+        {
+          url: "https://exokit.org/",
+          title: "Home",
+          active: window.location.host.split(".")[0] === "exokit"
+        }
+      ]
+
+      // Populate nav-popout with links
+      document.getElementById('nav-popout').innerHTML = navLinks.map((link, index) => {
+        return `
+          <div>
+            <a href=${link.url} class=${link.active ? "activeNav" : ""}>${link.title}</a>
+          </div>
+        `
+      }).join(`<br/>`)
+
+      // Close nav-popout if user clicks away
+      document.addEventListener("click", (e) => {
```

Comment:
This could probably be done with a `blur` event and making the popout focusable (`tabindex=-1`).

Reply:
Reason being there are other reasons to close a popup, such as the user pressing tab, or something else popping up in front.

Reply:
👍